src/Workspaces/Core/Portable/Workspace/Workspace.cs

```diff
@@ -375,6 +376,14 @@ protected virtual void Dispose(bool finalize)
 
             (_optionService as IWorkspaceOptionService)?.OnWorkspaceDisposed(this);
             _optionService.UnregisterWorkspace(this);
+
+            // Directly dispose IRemoteHostClientProvider if necessary. This is a test hook to ensure RemoteWorkspace
+            // gets disposed in unit tests as soon as TestWorkspace gets disposed. This would be superseded by direct
+            // support for IDisposable in https://github.com/dotnet/roslyn/pull/47951.
+            if (Services.GetService<IRemoteHostClientProvider>() is IDisposable disposableService)
```

Comment:
📝 My one concern here is what happens if this triggers the JIT to try and load one of our optional dependencies which isn't relevant for typical command line `Workspace` usage. I couldn't find any direct indication that this would occur, but the possibility leaves me with a preference towards #47951 which is a much cleaner approach.